### PR TITLE
Feat/21 image data crud

### LINKED ISSUE #21: Image labelling data (Feature)
**Link to User Story**
TBD (1005)

**Feature Description**
The admin (customer) wants to upload, read(, update) and delete a set of image data for a labelling project.

**Additional context**
None

### PR DESCRIPTION
**Pull request by guentgia**
- implement image processing
- upload single image files (jpg, jpeg, png) or multiple compressed in a zip file
- other smaller adjustments to ProjectController and DataPointController
- Tests for image processing API endpoints
closes #21

### CODE REVIEW

File: D-lama-service/Data/ProjectEntities/Project.cs [hunk 1]
```diff
@@ -42,7 +45,17 @@ public Project(string name, string description)
             Name = name;
             Description = description;
             UpdateDate = DateTime.UtcNow;
-            IsReady = false;
+            DataType = "text";
```
Comment (joelgrand) [inline D-lama-service/Data/ProjectEntities/Project.cs:48]: Maybe it is easier to use an Enum here, as you can then make comparison using the Enum later on:

if(project.DataType == ProjectType.Text){...}, so you don't have the coupling of the explicit string value.

just as an example.

**Review by joelgrand — CHANGES_REQUESTED** [review]
Hey Gianbello,

I had a quick look and for me it looks really great! Nice work! I just added one improvement idea... if you disagree, you can already go on merge it as the rest seems to be very nicely implemented! Thanks!!

File: D-lama-service/Data/ProjectEntities/Project.cs (hunk 1 above)
Comment (guentgia) [reply D-lama-service/Data/ProjectEntities/Project.cs]: That's a great advice. I forgot about enums completely!

**Comment by guentgia** [comment]
Added enum type für project data type: 0 = text, 1 = image

**Review by joelgrand — APPROVED** [review]
Perfect! Thank you Gianbello

**Pull request merged**